src/Cafe/IOSU/legacy/iosu_fpd.cpp

```diff
@@ -214,6 +214,13 @@ namespace iosu
 			friendData->friendExtraData.gameKey.ukn08 = frd->presence.gameKey.ukn;
 			NexPresenceToGameMode(&frd->presence, &friendData->friendExtraData.gameMode);
 
+			auto fixed_presence_msg = '\0' + frd->presence.msg; // avoid first character of comment from being cut off
+			friendData->friendExtraData.gameModeDescription.assignFromUTF8(fixed_presence_msg);
+			
+			auto comment_utf16 = StringHelpers::FromUtf8(frd->comment.commentString);
+			comment_utf16.insert(0, 1, '\0'); // avoid first character of comment from being cut off
+			memcpy(friendData->friendExtraData.comment, comment_utf16.c_str(), 36);
```

Comment:
We have `friendExtraData.comment` defined as a char array, but here you are assigning it the bytes from a wide string. If this is working then we should redefine `friendExtraData.comment` as the proper string type. Consider using the helper class `CafeWideString` since that avoids having to manually memcpy